GUI/Dialogs/EditLabelsDialog.cs

```diff
@@ -275,6 +276,8 @@ private bool TrySave(out string errMsg)
                 {
                     labels.Labels = labels.Labels
                         .Concat(new ModuleLabel[] { currentlyEditing })
+                        .OrderBy(l => l.InstanceName == null)
+                        .ThenBy(l => l.InstanceName)
                         .ToArray();
```

Comment:
This doesn't work for newly created instance-specific labels (or those moved from global to instance or vice versa), they are still sorted below global labels because when this sorting logic is run, the label doesn't have `InstanceName` assigned yet. That happens 4 lines later.

If I'm not mistaken, we should be fine moving the whole ` if(!labels.Labels.Contains(currentlyEditing))` block right above the `EditDetailsPanel.Visible = false;` line, so that the new label is completely configured before we sort it into the array.

Reply:
Yes, I think you're right. Change made.